drools-ansible-rulebook-integration-api/src/test/java/org/drools/ansible/rulebook/integration/api/toexecmodel/ToExecModelRulesOnceAfterTest.java

```diff
@@ -0,0 +1,101 @@
+import org.drools.ansible.rulebook.integration.api.domain.conditions.MapCondition;
+import org.drools.ansible.rulebook.integration.api.domain.temporal.OnceAfterDefinition;
+import org.drools.ansible.rulebook.integration.api.domain.temporal.Throttle;
+import org.drools.ansible.rulebook.integration.api.rulesengine.RulesExecutionController;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.drools.ansible.rulebook.integration.api.rulesengine.RegisterOnlyAgendaFilter.RULE_TYPE_TAG;
+import static org.drools.ansible.rulebook.integration.api.rulesengine.RegisterOnlyAgendaFilter.SYNTHETIC_RULE_TAG;
+import static org.drools.ansible.rulebook.integration.api.utils.TestUtils.createEventField;
+import static org.drools.ansible.rulebook.integration.api.utils.TestUtils.createSingleMap;
+import static org.drools.ansible.rulebook.integration.api.utils.TestUtils.getRuleByName;
+
+public class ToExecModelRulesOnceAfterTest extends ToPatternTestBase {
+
+    @Test
+    public void onceAfter() throws Exception {
+        // Rule
+        Rule rule = new Rule();
+        rule.setName("r1");
+
+        // Condition
+        LinkedHashMap<Object, Object> lhsValueMap = createEventField("alert.level");
+        LinkedHashMap<Object, Object> rhsValueMap = createSingleMap("String", "warning");
+        LinkedHashMap<Object, Object> equalsExpression = createEqualsExpression(lhsValueMap, rhsValueMap);
+        LinkedHashMap<Object, Object> rootMap = createAllCondition(equalsExpression);
+        MapCondition mapCondition = new MapCondition(rootMap);
+        rule.setCondition(mapCondition);
+
+        // Action
+        LinkedHashMap<Object, Object> actionDetails = createSingleMap("action", "print_event");
+        MapAction mapAction = new MapAction();
+        mapAction.put("Action", actionDetails);
+        rule.setAction(mapAction);
+
+        // OnceAfter
+        Throttle throttle = new Throttle();
+        throttle.setOnce_after("10 seconds");
+        throttle.setGroup_by_attributes(List.of("event.meta.host", "event.alert.level"));
+        rule.setThrottle(throttle);
+
+        // RulesSet
+        RulesSet rulesSet = new RulesSet();
+        RuleContainer ruleContainer = new RuleContainer();
+        ruleContainer.setRule(rule);
+        rulesSet.setRules(List.of(ruleContainer));
+
+        RulesExecutionController rulesExecutionController = new RulesExecutionController();
+        AtomicInteger ruleCounter = new AtomicInteger(0);
+
+        List<org.drools.model.Rule> rules = rule.toExecModelRules(rulesSet, rulesExecutionController, ruleCounter);
+
+        assertThat(rules).as("OnceAfter generates additional 3 synthetic rules").hasSize(4);
+
+        // r1
+        org.drools.model.Rule r1 = getRuleByName(rules, "r1");
+        assertThat(r1).isNotNull();
+        assertThat(r1.getMetaData(RULE_TYPE_TAG)).isEqualTo(OnceAfterDefinition.KEYWORD);
```

Comment:
predicate tests would be too messy and consequences cannot be tested, so I resulted in these simple assertions. We would mostly rely on integration tests like `OnceAfterTest`, but a finer grained test like this is a little good to have.